src/safeposix/syscalls/fs_calls.rs

```diff
@@ -4111,60 +4144,121 @@ impl Cage {
         let metadata = &SHM_METADATA;
         let is_shared = pshared != 0;
 
-        // Iterate semaphore table, if semaphore is already initialzed return error
+        // Iterate semaphore table, if semaphore is already initialized return error & Will initialize only it's new
```

Comment:
This comment isn't really clear, though it definitely wasn't clear from whoever wrote it initially (might be me! oops!) What is actually happening here?